### PR TITLE
build: update bazel rules_nodejs to 2.3.1 to use the fixed `npm_package.pack` rule on Windows OS

### PR DESCRIPTION
**Pull request by JiaLiPassion**
Update `rules_nodejs` to 2.3.1 to use the new `npm_package.pack` rule which is also working on `Windows OS` now by this fix https://github.com/bazelbuild/rules_nodejs/commit/bc36519

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
LGTM.

// @petebacondarwin fyi since you mentioned rules_nodejs today.

**Comment by petebacondarwin** [comment]
> LGTM.
> 
> // @petebacondarwin fyi since you mentioned rules_nodejs today.

Thanks @IgorMinar - I was told v3 but perhaps they meant v2.3? I'll give it a try

**Review by gkalpak — APPROVED** [review]
:tada:

**Review by gkalpak — CHANGES_REQUESTED** [review]
Actually, I just noticed that running `node scripts\build\build-packages-dist.js` leaves behind an `undefined-0.1.0.tgz` file in my working directory 😕

**Comment by JiaLiPassion** [comment]
@gkalpak , yeah, I fixed it in this PR https://github.com/bazelbuild/rules_nodejs/pull/2282#event-3985627948, do we need to wait for the next release?

**Comment by gkalpak** [comment]
Nice! I would say let's wait for the next release, because people will see this bogus `.tgz` there and won't know what to do with it or where it came from.

**Comment by JiaLiPassion** [comment]
@gkalpak , got it, thanks!

**Comment by JiaLiPassion** [comment]
@gkalpak , the `rule_nodejs` has been updated to `3.5.1`, and the issue earlier `undefined-0.1.0.tgz` is gone, could you take a look again? Thanks.

File: integration/bazel/WORKSPACE [hunk 1]
```diff
@@ -5,8 +5,8 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 # Fetch rules_nodejs so we can install our npm dependencies
 http_archive(
     name = "build_bazel_rules_nodejs",
-    sha256 = "4952ef879704ab4ad6729a29007e7094aef213ea79e9f2e94cbe1c9a753e63ef",
-    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/2.2.0/rules_nodejs-2.2.0.tar.gz"],
+    sha256 = "452bef42c4b2fbe0f509a2699ffeb3ae2c914087736b16314dbd356f3641d7e5",
+    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/2.3.0/rules_nodejs-2.3.0.tar.gz"],
```
Comment (gkalpak) [inline integration/bazel/WORKSPACE:9]: Shouldn't we use v2.3.1 here too:

```suggestion
    sha256 = "121f17d8b421ce72f3376431c3461cd66bfe14de49059edc7bb008d5aebd16be",
    urls = ["https://github.com/bazelbuild/rules_nodejs/releases/download/2.3.1/rules_nodejs-2.3.1.tar.gz"],
```

**Review by gkalpak — APPROVED** [review]
:+1:

**Comment by alxhub** [comment]
@JiaLiPassion this PR doesn't merge cleanly to 11.0.x. Please open a separate PR for that branch if needed.

**Pull request closed without merging**